src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.Small.cs

```diff
@@ -58,7 +58,7 @@ protected override bool ReadAndCacheConstructorArgument(ref ReadStack state, ref
                     }
                     break;
                 default:
-                    Debug.Fail("This should never happen.");
+                    Debug.Fail("We shouldn't be here if there are more than 4 parameters.");
```

Comment:
What is "here" here?  You mean in this method?

Reply:
Yes, we should be in the method override for LargeObjectWithParameterizedConstructorConverter. I've added more detail in the message..